src/main.rs

```diff
@@ -1,58 +1,45 @@
-// src/main.rs
+mod node;
+mod routing_table;
+mod kbucket;
+mod message;
+mod utils;
+use log::{info, warn, debug, error};
 
-use kademlia_mvp::*;
-use tokio::time::Duration;
-use tokio::time::sleep;
-use log::{debug, error, info, warn};
+use node::KademliaNode;
+// use tokio::net::SocketAddr;
+use std::net::SocketAddr;
 
 #[tokio::main]
-async fn main() -> Result<(), Box<dyn std::error::Error>> {
-    // Initialize logging
-    env_logger::init();
+async fn main() -> std::io::Result<()> {
+    env_logger::init(); // Initialize the logger
+    info!("Starting Kademlia node...");
 
-    // Create two nodes
-    let addr1 = "127.0.0.1:8001".parse()?;
-    let addr2 = "127.0.0.1:8002".parse()?;
-    let mut node1 = KademliaNode::new(addr1).await;
-    let node2 = KademliaNode::new(addr2).await;
+    let addr: SocketAddr = "127.0.0.1:33334".parse().expect("Failed to parse socket address");
```

Comment:
at some point we might want to introduce configurations and make this configurable.